libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/AndroidSpeechPlayer.java

```diff
@@ -52,10 +53,14 @@ public void onInit(int status) {
   /**
    * Plays the given voice instruction using TTS
    *
-   * @param instruction voice instruction to be synthesized and played
+   * @param speechAnnouncement with voice instruction to be synthesized and played
    */
   @Override
-  public void play(String instruction) {
+  public void play(SpeechAnnouncement speechAnnouncement) {
+    if (speechAnnouncement == null) {
```

Comment:
I would add this in the boolean `canPlay`. i.e., `boolean canPlay = languageSupported && !isMuted && !TextUtils.isEmpty(instruction) && speechAnnouncement != null;`